src/token/colmetadata-token-parser.ts

```diff
@@ -56,48 +57,230 @@ function readColumnName(parser: Parser, options: InternalConnectionOptions, inde
+    return callback();
+  }
 
-    let i = 0;
-    function next(done: () => void) {
-      if (i === columnCount) {
-        return done();
-      }
+  // 2.2.7.4 COLMETADATA
+  readCryptoMetadataOrdinal(parser, columnEncryptionKeyTables, (ordinal) => {
+    metadataParseCrypto(parser, options, (baseTypeMetadata) => {
+      parser.readUInt8((cipherAlgorithmId) => {
+        readCustomEncryptionMetadata(parser, cipherAlgorithmId, (cipherAlgorithmName) => {
+          parser.readUInt8((encryptionType) => {
+            parser.readBuffer(1, (normalizationRuleVersion) => {
+              callback({
+                ordinal,
+                cipherAlgorithmId,
+                cipherAlgorithmName,
+                encryptionType,
+                normalizationRuleVersion,
+                baseTypeInfo: baseTypeMetadata,
+                encryptionKeys: columnEncryptionKeyTables ? columnEncryptionKeyTables[ordinal] : undefined,
+              });
+            });
+          });
+        });
+      });
+    });
+  });
+}
+
+function readCustomEncryptionMetadata(parser: Parser, cipherAlgorithmId: number, callback: (cipherAlgorithmName?: string) => void) {
+  // 2.2.6.6 RPC Request
+  // This byte describes the encryption algorithm that is used. For a custom encryption algorithm,
+  // the EncryptionAlgo value MUST be set to 0 and the actual encryption algorithm MUST be
+  // inferred from the AlgoName. For all other values, AlgoName MUST NOT be sent.
+  // If the value is set to 1, the encryption algorithm that is used is
+  // AEAD_AES_256_CBC_HMAC_SHA512
+  if (cipherAlgorithmId !== 0) {
+    return callback();
+  }
+
+  parser.readUInt8((nameSize) => {
+    parser.readBuffer(nameSize, (algorithmNameBuffer) => {
+      callback(algorithmNameBuffer.toString('ucs2'));
+    });
+  });
+}
+
+function readCryptoMetadataOrdinal(parser: Parser, columnEncryptionKeys: EncryptionKeyInfo[][] | undefined, callback: (ordinal: number) => void) {
+  if (!columnEncryptionKeys) {
+    // Ordinal is not provided for SQL function return values
+    callback(0);
+  } else {
+    parser.readUInt16LE((ordinal) => {
+      callback(ordinal);
+    });
+  }
+}
+
+function readEncryptedCEK(parser: Parser, callback: (encryptedCEK: Buffer) => void) {
+  parser.readUInt16LE((encryptedCEKLength) => {
+    parser.readBuffer(encryptedCEKLength, (encryptedCEK) => {
+      callback(encryptedCEK);
+    });
+  });
+}
+
+function readKeyStoreName(parser: Parser, callback: (keyStoreName: string) => void) {
+  parser.readUInt8((keyStoreNameLength) => {
+    parser.readBuffer(2 * keyStoreNameLength, (keyStoreNameBuffer) => {
+      callback(keyStoreNameBuffer.toString('ucs2'));
+    });
+  });
+}
+
+function readKeyPath(parser: Parser, callback: (keyPath: string) => void) {
+  parser.readUInt8((keyPathLength) => {
+    parser.readBuffer(2 * keyPathLength, (keyPathBuffer) => {
+      callback(keyPathBuffer.swap16().toString('ucs2'));
+    });
+  });
```

Comment:
```suggestion
  parser.readBVarChar(callback);
```

Reply:
`parser.readBVarChar(callback);` doesn't do the `swap16` needed here